testing/integration/src/subscribe_benchmarks.rs

```diff
@@ -0,0 +1,305 @@
+const SUBSCRIBE_WORKERS: usize = 20;
+
+#[cfg(feature = "heap")]
+const MAX_MEMORY: u64 = 22_000_000_000;
+#[cfg(not(feature = "heap"))]
+const MAX_MEMORY: u64 = 31_000_000_000;
+
+const NOTIFY_CLIENTS: usize = 500;
+const MAX_ADDRESSES: usize = 1_000_000;
+const WALLET_ADDRESSES: usize = 800;
+
+const STAT_FOLDER: &'static str = "../../../analyze/mem-logs";
+
+fn create_client_addresses(index: usize, network_id: &NetworkId) -> Vec<Address> {
+    // Process in heaviest to lightest requests order, maximizing messages memory footprint
+    // between notifiers and from notifier to broadcasters at grpc server and rpc core levels
+    let max_address = ((NOTIFY_CLIENTS - index) * MAX_ADDRESSES / NOTIFY_CLIENTS) + 1;
+    let min_address = if (NOTIFY_CLIENTS - index) % (NOTIFY_CLIENTS / 5) == 0 {
+        // Create a typical UTXOs monitoring service subscription scope
+        0
+    } else {
+        // Create a typical wallet subscription scope
+        max_address.max(WALLET_ADDRESSES) - WALLET_ADDRESSES
+    };
+    (min_address..max_address)
+        .map(|x| Address::new((*network_id).into(), kaspa_addresses::Version::PubKey, &Uint256::from_u64(x as u64).to_le_bytes()))
+        .collect_vec()
+}
+
+/// `cargo test --package kaspa-testing-integration --lib --features devnet-prealloc -- subscribe_benchmarks::utxos_changed_subscriptions_sanity_check --exact --nocapture --ignored`
+#[tokio::test]
+#[ignore = "bmk"]
+async fn utxos_changed_subscriptions_sanity_check() {
+    init_allocator_with_default_settings();
+    kaspa_core::panic::configure_panic();
+    kaspa_core::log::try_init_logger(
+        "INFO, kaspa_core::time=debug, kaspa_rpc_core=debug, kaspa_grpc_client=debug, kaspa_notify=info, kaspa_notify::address::tracker=debug, kaspa_notify::listener=debug, kaspa_notify::subscription::single=debug, kaspa_mining::monitor=debug, kaspa_testing_integration::subscribe_benchmarks=trace", 
+    );
+
+    let (prealloc_sk, _) = secp256k1::generate_keypair(&mut thread_rng());
+    let args = ArgsBuilder::simnet(TX_LEVEL_WIDTH as u64 * CONTRACT_FACTOR, PREALLOC_AMOUNT)
+        .apply_args(|args| Daemon::fill_args_with_random_ports(args))
+        .build();
+
+    // Start the daemon
+    info!("Launching the daemon...");
+    let daemon_args = DaemonArgs::new(
+        args.rpclisten.map(|x| x.normalize(0).port).unwrap(),
+        args.listen.map(|x| x.normalize(0).port).unwrap(),
+        prealloc_sk.display_secret().to_string(),
+        Some("ucs-server".to_owned()),
+    );
+    let server_start_time = std::time::Instant::now();
+    let mut daemon_process = tokio::process::Command::new("cargo")
+        .args(daemon_args.to_command_args("subscribe_benchmarks::bench_utxos_changed_subscriptions_daemon"))
+        .spawn()
+        .expect("failed to start daemon process");
+
+    // Make sure that the server was given enough time to start
+    let client_start_time = server_start_time + Duration::from_secs(DAEMON_LAUNCH_SECS_DELAY);
```

Comment:
It's a child process, so we have no access to its internal memory from the parent process.

There might be some fancy interprocess data sharing tricks that we might implement but it seems overkill here.

Just relying on RPC giving some time so the server responds seems ok, especially since the test is ignored for belonging to a benchmark, thus not executed either locally with nextest or in CI.